info-compiler/src/main/java/com/google/sps/webcrawler/RelevancyChecker.java

```diff
@@ -50,20 +53,27 @@ public RelevancyChecker() throws IOException {
 
   /**
    * Checks whether the {@code newsArticle} is relevant to the {@code candidateName} of interest.
-   * Defines relevancy as the salience of {@code candidateName} in the content, and defines
-   * sufficient relevancy with {@code SALIENCE_THRESHOLD}.
+   * Defines relevancy as the salience of {@code candidateName} and {@code partyName} in the
+   * content both being bigger than their respective threshold. If {@code partyName} is null,
+   * skips the salience checking for {@code partyName} and determines relevancy solely by looking
+   * at the salience of {@code candidateName}.
    */
-  public boolean isRelevant(NewsArticle newsArticle, String candidateName) {
-    double salience = computeSalienceOfName(newsArticle.getContent(), candidateName);
-    return salience >= SALIENCE_THRESHOLD;
+  public boolean isRelevant(NewsArticle newsArticle, String candidateName, String partyName) {
+    double candidateNameSalience = computeSalienceOfName(newsArticle.getContent(), candidateName);
+    double partyNameSalience =
+        (partyName == null || partyName.equalsIgnoreCase(NO_PARTY_AFFILIATION)
+             || partyName.equalsIgnoreCase(NON_PARTISAN) || partyName.equalsIgnoreCase(NONPARTISAN))
+        ? PARTY_SALIENCE_THRESHOLD : computeSalienceOfName(newsArticle.getContent(), partyName);
+    return (candidateNameSalience >= CANDIDATE_SALIENCE_THRESHOLD
+        && partyNameSalience >= PARTY_SALIENCE_THRESHOLD);
```

Comment:
The way this is structured (returning a number that is exactly the threshold when you want to ignore party salience) is a little bit confusing. Maybe instead of using a ternary operator here you can just structure the whole block to use an if-else statement?

Reply:
Ok I'll make the change! Thanks